op-chain-ops/setup/deploy/l2_genesis.go

```diff
@@ -0,0 +1,60 @@
+package deploy
+
+import "github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+
+// FullL2 defines how to create a full L2 genesis, no splits in work
+type FullL2 struct {
+	Args struct {
+		genesis.L2InitializationConfig
+		genesis.L1DependenciesConfig
+	}
+
+	Addresses struct{} // No dependencies in L2
+}
+
+func (cfg *FullL2) ScriptTarget() string {
+	return "L2Genesis.s.sol"
+}
+
+func (cfg *FullL2) ScriptSig() string {
+	return "l2Genesis"
+}
+
+func (cfg *FullL2) ScriptDependencies() []string {
```

Comment:
I am a bit worried about having to list contract names here as it creates this leaky abstraction between solidity and go. Could the `ScriptDependencies` be other Go `Deployment`s (not sure what interface name is) ?